pkg/scaling/scale_handler.go

```diff
@@ -163,6 +169,58 @@ func (h *scaleHandler) startScaleLoop(ctx context.Context, withTriggers *kedav1a
 	}
 }
 
+func (h *scaleHandler) GetScalersCacheForScaledObject(ctx context.Context, scaledObjectName, scaledObjectNamespace string) (*cache.ScalersCache, error) {
+	key := kedav1alpha1.GenerateIdenitifier("ScaledObject", scaledObjectName, scaledObjectNamespace)
+
+	h.lock.RLock()
+	if cache, ok := h.scalerCaches[key]; ok {
+		h.lock.RUnlock()
+		return cache, nil
+	}
+	h.lock.RUnlock()
+
+	h.lock.Lock()
+	defer h.lock.Unlock()
+	if cache, ok := h.scalerCaches[key]; ok {
+		return cache, nil
+	} else if ok {
+		cache.Close(ctx)
+	}
```

Comment:
why is this different from the if above? I mean, this condition will be always false because if it's true, the flow will return with the if above.
Apart from that, both branches test the same `ok`, so logically this won't happen never I'd say

Reply:
good catch, this is just a bad copy paste